com.unity.visualeffectgraph/Editor/Compiler/VFXCodeGenerator.cs

```diff
@@ -362,10 +362,16 @@ static private void SubstituteMacros(StringBuilder builder)
             blockCallFunctionContent = blockCallFunction.builder.ToString();
         }
 
-        internal static void BuildParameterBuffer(VFXContextCompiledData contextData, out string parameterBufferContent)
+        internal static void BuildParameterBuffer(VFXContextCompiledData contextData, IEnumerable<string> filteredOutTextures, out string parameterBufferContent)
```

Comment:
I find that strange to have a parameter called filteredOutTextures that end up being written in the function...

Reply:
This helper function is really specific to the new SG integration.
I can potentially move the logic from https://github.com/Unity-Technologies/Graphics/pull/5416/files#diff-7174c990ca8df27503bf5973a949e19bc00cf239832d30b20b2a69167667272eR288 into this "BuildParameterBuffer".
But afterwards, it would be the same, the `WriteTextures` function have a filter parameter...

The problem here is to handle texture only used by the VFX, the textures used in shaderGraph will be declared by shaderGraph generation.

Alternatively, I could have alter the shaderGraph code generation behavior to remove the exposed texture properties (admitting they are declared by vfx)